### PR TITLE
Add complex-input support to jax.scipy.special.gamma

### PR DESCRIPTION
**Pull request by ahmedtaha100**
Continuation of #36471 (closed when I accidentally deleted my fork). All review feedback from @jakevdp has been addressed.

**Implementation** (`jax/_src/scipy/special.py`):

- `_complex_loggamma(z)`: principal-branch log-gamma via Lanczos approximation (g=7, N=9) with reflection formula for Re(z) < 0.5. Safe masking prevents NaN gradient contamination through `jnp.where`'s VJP.
- `_complex_gamma(z)`: wrapper computing `exp(loggamma(z))` with pole and NaN handling.
- `gamma()` dispatches on dtype: real inputs use the existing `lgamma` path, complex inputs use the new Lanczos implementation.
- Poles at non-positive integers return `nan+nanj` (matching SciPy).

**Review feedback addressed** (from #36471):

1. Lanczos coefficients cited to GSL `specfunc/gamma.c` and Lanczos 1964 paper
2. Removed dtype promotion — computes at input precision
3. Added assert for `complexfloating`
4. Vectorized Lanczos sum with `jnp.sum` + `lax.expand_dims` (no implicit rank promotion)
5. `jnp.isnan(z)` instead of checking real/imag separately
6. `jnp.nan` instead of `float('nan')`
7. Scalar `nan_val` instead of full vector
8. Condensed `_complex_gamma` to single `jnp.where` with safe pre-masking
9. Tests reduced to `JAX_SPECIAL_FUNCTION_RECORDS` entry + 2 targeted tests

**Tests** (`tests/lax_scipy_special_functions_test.py`):

- `gamma_complex` entry in `JAX_SPECIAL_FUNCTION_RECORDS` for SciPy parity
- `testComplexGammaPoles`: poles at non-positive integers
- `testComplexGammaBranchCut`: conjugate symmetry near Re(z) = 0.5

All 420 tests pass locally under `JAX_ENABLE_X64=1`.

Addresses #6765

### CODE REVIEW

File: tests/lax_scipy_special_functions_test.py [hunk 1]
```diff
@@ -414,6 +418,23 @@ def testSiciRaiseOnComplexInput(self):
     with self.assertRaisesRegex(ValueError, "Argument `x` to sici must be real-valued."):
       lsp_special.sici(samples)
 
+  def testComplexGammaPoles(self):
+    """Test that gamma returns nan+nanj at non-positive integer poles."""
+    with jax.enable_x64():
```
Comment (jakevdp) [inline tests/lax_scipy_special_functions_test.py]: Please don't enable x64 like this in the test – every test is run in both X32 and X64 precision. If for some reason it fails in X32, it would be better to skip in that case. (but that brings up the larger question: why does it fail in X32, and is there something we can do to address that failure?)

File: tests/lax_scipy_special_functions_test.py [hunk 2]
```diff
@@ -414,6 +418,23 @@ def testSiciRaiseOnComplexInput(self):
...
+      poles = jnp.array([0+0j, -1+0j, -2+0j, -5+0j], dtype=jnp.complex128)
+      result = np.array(lsp_special.gamma(poles))
+      # Both real and imaginary parts should be NaN
+      self.assertTrue(np.all(np.isnan(result.real)))
+      self.assertTrue(np.all(np.isnan(result.imag)))
+
+  def testComplexGammaBranchCut(self):
+    """Test gamma near the negative real axis and at the reflection boundary."""
+    with jax.enable_x64():
```
Comment (jakevdp) [inline tests/lax_scipy_special_functions_test.py]: Same here.

**Review by jakevdp — CHANGES_REQUESTED** [review]

File: tests/lax_scipy_special_functions_test.py (hunk 1 above)
Comment (ahmedtaha100) [reply tests/lax_scipy_special_functions_test.py]: This is great feedback, thank you. Removed forced x64 from both tests, they now run in both x32 and x64. The only follow up change was relaxing the branch cut tolerance from 1e-10 to 1e-5, since 1e-10 is too strict for complex64 (observed error is around 1e-6 in x32). The poles test needed no tolerance change because it only checks `np.isnan`

**Review by jakevdp — APPROVED** [review]
Looks good! Last thing: could you squash the changes into a single commit now that everything is ready to go? Thanks!

**Comment by ahmedtaha100** [comment]
> Looks good! Last thing: could you squash the changes into a single commit now that everything is ready to go? Thanks!

Done! Squashed into a single commit.

**Review by jakevdp — APPROVED** [review]

**Review by jakevdp — CHANGES_REQUESTED** [review]
It looks like we have some TPU failures in `testComplexGammaBranchCut`.

We'll need to adjust the tolerance when running on TPU – you can use the pattern seen elsewhere in the file; something like this:
```
rtol = 1E-3 if jtu.test_device_matches(["tpu"]) else 1e-5
```

**Comment by ahmedtaha100** [comment]
> It looks like we have some TPU failures in `testComplexGammaBranchCut`.
> 
> We'll need to adjust the tolerance when running on TPU – you can use the pattern seen elsewhere in the file; something like this:
> 
> ```
> rtol = 1E-3 if jtu.test_device_matches(["tpu"]) else 1e-5
> ```

Thank you for your help. I added a TPU specific rtol guard in testComplexGammaBranchCut to match existing patterns in this file (1E-3 on TPU, 1e-5 otherwise) and address the TPU failures. I pushed this as an incremental commit to make review easier, if this looks good, I’ll squash back to a single commit.

**Comment by jakevdp** [comment]
Thanks for the test fix – everything looks good! Can you squash the commits again? After that we should be able to get this merged.

**Comment by ahmedtaha100** [comment]
> Thanks for the test fix – everything looks good! Can you squash the commits again? After that we should be able to get this merged.

Great, just squashed into a single commit. Please let me know if there's anything else needed!

**Review by jakevdp — APPROVED** [review]

**Comment by jakevdp** [comment]
Thanks for all your work on this!

**Comment by ahmedtaha100** [comment]
> Thanks for all your work on this!

Thank you for all the detailed feedback throughout this PR, the reviews were incredibly helpful and I learned a lot from the process. Really appreciate you taking the time. I look forward to contributing more!
